Userland/Services/SystemServer/Service.cpp

```diff
@@ -56,13 +56,16 @@ Service* Service::find_by_pid(pid_t pid)
 void Service::set_enabled_for_current_boot_mode(bool enabled)
 {
     extern String g_boot_mode;
-    
+
     if (!is_enabled() && enabled) {
         m_boot_modes.append(g_boot_mode);
     } else {
         m_boot_modes.remove_first_matching([&](auto& mode) { return mode == g_boot_mode; });
     }
     dbgln("Service::set_enabled_for_current_boot_mode({}) {}: {}", enabled, g_boot_mode, m_boot_modes.size());
+
+    auto config = Core::ConfigFile::get_for_system("SystemServer");
+    config->write_entry(name(), "BootModes", String::join(',', m_boot_modes));
```

Comment:
I'm not sure I like the server modifying its own config file. Do our other servers do this?

A config file sounds like something the server reads, but never writes. It's the administrator who edits the file. Perhaps the `service` utility should write to the file directly, and then notify SystemServer about the changes? (That is also how `systemctl edit` works.)